File: mobile/src/services/onboarding.ts
import { apiRequest } from './api';
import type { User } from './auth';

export interface OnboardingData {
  display_name?: string;
  monthly_income: string;
  salary_day: number;
  currency: string;
  total_budget: string;
  category_limits: Array<{ category_id: string; limit_amount: string }>;
}

export async function completeOnboarding(data: OnboardingData): Promise<User> {
  return apiRequest<User>('/api/v1/onboarding/complete', {
    method: 'POST',
    body: JSON.stringify(data),
  });
}

File: backend/app/schemas/onboarding.py
from decimal import Decimal

from pydantic import BaseModel, Field


class OnboardingRequest(BaseModel):
    display_name: str | None = None
    monthly_income: Decimal = Field(gt=0)
    salary_day: int = Field(ge=1, le=31)
    currency: str = Field(default="INR", max_length=3)
    total_budget: Decimal = Field(gt=0)
    category_limits: list[dict] = Field(default_factory=list)


class ProfileUpdateRequest(BaseModel):
    display_name: str | None = None
    monthly_income: Decimal | None = Field(default=None, gt=0)
    salary_day: int | None = Field(default=None, ge=1, le=31)
    currency: str | None = Field(default=None, max_length=3)
